### ISSUE TITLE
Add a code friendly list of supported methods to the package

### ISSUE THREAD

**Issue by sramam** [faker-js/faker#1210]
### Clear and concise description of the problem

When `faker.js` is integrated with other tools, it's vey convenient to have a list of methods. This can be used for things like 
- providing "did you mean" support  
- generating links to `faker.js` documentation. 

In earlier versions of faker.js this was possible to extract by introspecting the faker instance. Since the move to typedoc, this became a build time problem.

### Suggested solution

It's relatively trivial to generate a list of supported methods and include it in the package. 
Since it's just a list of names, it's light-weight and stays in sync with the package version.

### Additional context

I'd be happy to provide a pull request if this proposal is acceptable.

**Comment by Shinigami92** [comment]
We are thinking of providing a docs page that is searchable and behave like https://vuejs.org/api/
But the preconditions are that we want to wait on vitepress v1 release

Would this fit your need? Or are you proposing something differently?

**Comment by sramam** [comment]
thanks for taking the time to review this request. 
I was thinking more of a form that is programmatically accessible. 
Something along the lines of [this gist](https://gist.github.com/sramam/7372b1885083700277f0c6a847e8ff45).

**Comment by ST-DDT** [comment]
What do you want to use that list for?
AFAICT you don't want it for some TS feature, that proposes the methods.
Instead you want to just have it for some program?
Note you can still introspect the faker instance as before (we partially do that in our tests).
https://github.com/faker-js/faker/blob/a65f1a2b59b909c4b5cc044af560ff91642c9a56/test/support/seededRuns.ts#L284-L287

**Comment by sramam** [comment]
Ah interesting. 
Would this be the right way to get the list of supported modules then?

```
const modules = 
  Object.entries(faker)
    .filter(([key, val]) => 
      typeof val === 'object' && 
      key !== 'definitions')
```

I need the list of supported faker values to be able to provide some debug information in my program, and would like a way to upgrade the faker.js version in an automated fashion. Hence the desire to get to a "master list" of methods being supported programatically.

**Comment by ST-DDT** [comment]
Probably this:

https://github.com/faker-js/faker/blob/aedd7b8bb0e6074f69659a79df1e6b75900f5894/test/all_functional.spec.ts#L49-L65

**Comment by sramam** [comment]
ah. Very kind of you. 

One more question:

Is there a way to pull out deprecated function calls too? 
Given that my code and its use is green-field, better to prevent use of deprecated methods.

**Comment by sramam** [comment]
> I did not mean for this to turn into a support request. Please let me know if I should be taking this elsewhere. 

This is the function I am using:

```
function getSupportedFakerMethods(): string[] {
  // https://github.com/faker-js/faker/blob/aedd7b8bb0e6074f69659a79df1e6b75900f5894/test/all_functional.spec.ts#L4
  const IGNORED_MODULES = [
    "locales",
    "locale",
    "localeFallback",
    "definitions",
    "fake",
    "helpers",
    "mersenne",
  ];
  const methods = Object.entries(faker)
    .filter(([key, val]) => !IGNORED_MODULES.includes(key))
    .map(([module]) => {
      return Object.entries((faker as any)[module])
        .filter(([, value]) => typeof value === "function")
        .map(([key]) => `faker.${module}.${key}`);
    })
    .flat(10)
    .sort();
  return methods;
}
```

I am seeing some oddities (compared to the version posted in the gist which extracted information from the api docs). Some deprecated methods are showing up and other active methods are not.
```
+   "faker.address.cityPrefix",
+   "faker.address.citySuffix",
+   "faker.commerce.color",
+   "faker.company.companyName",
 -   "faker.company.name",

```

**Comment by ST-DDT** [comment]
`faker.company.companyName` is not yet deprecated in the latest release. (starting v7.4 it will be).
`faker.company.name` is not yet released (it is in main, but will only be released starting with 7.4).

**Comment by sramam** [comment]
@ST-DDT, thank you for your help.
My initial request was addressed and this request is no longer pertinent.
